lib/openhab/dsl/rules/automation_rule.rb

```diff
@@ -110,8 +110,10 @@ def create_queue(inputs)
         def extract_event(inputs)
           event = inputs&.dig("event")
           unless event
-            event = Struct.new(:event, :attachment, :command).new
-            event.command = inputs&.dig("command")
+            input_keys = inputs.to_h.keys.grep(/^[a-z_]+[a-z0-9_]*$/).grep_v("module") # only pick valid identifiers
+            event_members = %i[event attachment] | input_keys.map(&:to_sym)
+            event = Struct.new(*event_members).new
```

Comment:
ugh... Struct.new in non-initialization code again. not your fault, you don't need to fix.

Reply:
I've refactored it so extract_event only gets called once for each execution. Furthermore, I changed it from Struct to a custom Hash.

Reply:
Custom hash didn't work out so well. Enumerable#command took precedence over method_missing. Reverted back to Struct.